### ISSUE TITLE
rebuildconf failure php7 and imagick 3.4.rc2

### ISSUE THREAD

**Issue by lastsamurai26** [mkoppanen/imagick#128]
:) I think I need to write anything

https://bugs.php.net/bug.php?id=70899

used Version 

php 7.0.0
wget http://de2.php.net/get/php-7.0.0.tar.gz/from/this/mirror -O php-7.0.0.tar.gz
imagick 3.4.0RC2
wget http://pecl.php.net/get/imagick-3.4.0RC2.tgz

tar xvzf php-7.0.0.tar.gz
tar xvzf imagick-3.4.0RC2.tgz
mv imagick-3.4.0RC2 imagick 
cp imagick php-7.0.0/ext
cd..
rm configure
./buildconf --force

Output 
Forcing buildconf
Removing configure caches
buildconf: checking installation...
buildconf: autoconf version 2.69 (ok)
rebuilding configure
/usr/bin/m4:ext/imagick/config.m4:14: cannot open `$config_m4_dir/imagemagick.m4': No such file or directory
autom4te: /usr/bin/m4 failed with exit status: 1
rebuilding main/php_config.h.in
/usr/bin/m4:ext/imagick/config.m4:14: cannot open `$config_m4_dir/imagemagick.m4': No such file or directory
autom4te: /usr/bin/m4 failed with exit status: 1
autoheader: '/usr/bin/autom4te' failed with exit status: 1

-------

i Don't know why but this version of imagick works with php-5.6

**Comment by Danack** [comment]
@lastsamurai26 can you recommend a slim vagrant box for Debian 8.3 ?

I can't repro this on centos which is what I normally dev on.

> this version of imagick works with php-5.6

Someone cocked up the PHP conf files in one of the 7.0.0 RC versions.....because RC releases are great times to be experimenting with autoconf make files. :-P

**Comment by lastsamurai26** [comment]
i have 2 Root server and one vserver i can test it on the Vserver, if you mean that.

if i use this patch https://gist.github.com/reeze/89fd045f1d0960d2add3#file-more-compatible-patch-diff

then it works 

but what you mean with to experimenting the autoconf make files

**Comment by Danack** [comment]
Actually, why are you doing buildconf --force? Try just doing:

    phpize
    ./configure 

or `./configure --libdir=/usr/lib64` if Imagick needs a hand finding the ImageMagick libraries.

**Comment by lastsamurai26** [comment]
~/php-7.0.0 # phpize
Cannot find config.m4.
Make sure that you run '/usr/bin/phpize' in the top level source directory of the module


php-7.0.0/ext/imagick # phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226

**Comment by Danack** [comment]
and then `make install` ?

**Comment by lastsamurai26** [comment]
i don't now what you mean?

if a do make install in php-7.0.0/ext/imagick
then i get this messege 
"make: *** No rule to make target 'install'.  Schluss."

**Comment by Danack** [comment]
Okay, unless you really, really need it to be built statically, I recommend compile Imagick separately.

So just compile PHP by itself first, and then do the below to compile Imagick.

```
cd /tmp
wget http://pecl.php.net/get/imagick-3.4.0RC2.tgz
tar xvzf imagick-3.4.0RC2.tgz
cd imagick-3.4.0RC2
phpize
./configure
make install

```

It should copy the imagick.so file generated into the the correct directory, and then you just need to add `extension=imagick.so` to the appropriate php.ini file.

**Comment by lastsamurai26** [comment]
ok, you mean. I've already made and that works well but I wanted to conserve my work :). That's why I've always compile  with PHP together, but if this is not work more readily I'll compile separated it

**Comment by Danack** [comment]
I will try and get it compiling like how you're trying, but it will should be working again in 7.0.1 anyway, as a bug fix has been applied to the 7.0 branch, just not in time for the release of 7.0.0

**Comment by lastsamurai26** [comment]
That means that as of version 7.0.1 php this works again as I compile it?

thx :)

**Comment by Danack** [comment]
You should be able to. The branch that will become the 7.0.1 version is https://github.com/php/php-src/tree/PHP-7.0 and it should have the fix in, so you could test against that branch now, to see if the alleged fix does work.

**Comment by lastsamurai26** [comment]
:) the Fix works thx

**Comment by Danack** [comment]
Could you try using this branch https://github.com/mkoppanen/imagick/tree/m4_refactoring ?

It should be able to be compiled statically as an extension.....I think. Maybe? Autoconf files really confuse me tbh.....

**Comment by lastsamurai26** [comment]
works :)

with php7 (stable)

**Comment by Danack** [comment]
This should be fixed in release [3.4.0RC3](https://pecl.php.net/package/imagick/3.4.0RC3)
